unstash-different-dir/unstashDifferentDir.groovy

```diff
@@ -0,0 +1,35 @@
+// First we'll generate a text file in a subdirectory on one node and stash it.
+stage "first step on first node"
+
+// Run on a node with the "first-node" label.
+node('first-node') {
+    // Make the output directory.
+    sh "mkdir -p output"
+
+    // Write a text file there.
+    writeFile file: "output/somefile", text: "Hey look, some text."
+
+    // Stash that directory and file.
+    stash name: "first-stash", includes: "output/*"
+}
+
+// Next, we'll make a new directory on a second node, and unstash the original
+// into that new directory, rather than into the root of the biuld.
+stage "second step on second node"
+
+// Run on a node with the "second-node" label.
+node('second-node') {
+    // Make the directory we're unstashing to.
+    sh "mkdir -p first-stash"
```

Comment:
I tend to do mkdir -p even when I probably don't need to - but I'll think about tweaking this.